tests/encrypted_lvm.spec.ts

```diff
@@ -14,37 +15,69 @@ test.describe('The main page', () => {
     });
 
     test('Installation test with encrypted lvm file system', async ({ page }) => {
+        // optional actions done on the page
+        const actions = Object.freeze({
+          setProduct: Symbol("product"),
+          setMainPage: Symbol("main"),
+          done: Symbol("done")
+        });
+
+        // check for multiple texts in parallel, avoid waiting for timeouts
+        let action = await Promise.any([
+          page.getByText("Product selection").waitFor().then(() => actions.setProduct),
+          page.getByText(process.env.PRODUCTNAME).waitFor().then(() => actions.setMainPage),
+        ]);
+        
+        // optional product selection
+        if (action === actions.setProduct) {
+          await test.step("Select the product", async () => {
+            const productSelectionPage = new ProductSelectionPage(page);
+            await productSelectionPage.expectProductSelection();
+            await productSelectionPage.checkServerType();
+            await productSelectionPage.select();
+          });
+
+          action = await Promise.any([
+            page.getByText(process.env.PRODUCTNAME).waitFor().then(() => actions.setMainPage),
+            page.getByText("Root authentication set").waitFor().then(() => actions.done),
+          ]);
+        }
+
         const mainPage = new MainPage(page);
-        await mainPage.accessStorage();
+        if (action === actions.setMainPage) {
+          await test.step("set encrypted lvm file system", async () => {
+            await mainPage.accessStorage();
 
-        const storagePage = new StoragePage(page);
-        await storagePage.useEncryption();
+            const storagePage = new StoragePage(page);
```

Comment:
just out of curiosity, are you using vscode or other tool?

Reply:
normally use vim, just used vscode edit the code.

Reply:
we need to format it in the way that vscode does, as Playwright is Microsoft, it makes sense to maintain that format, so that tabulation comes from vim or vscode? could you please ensure vscode format style?